src/qt/optionsmodel.cpp

```diff
@@ -320,10 +320,14 @@ static ProxySetting ParseProxyString(const QString& proxy)
     if (proxy.isEmpty()) {
         return default_val;
     }
-    // contains IP at index 0 and port at index 1
-    QStringList ip_port = GUIUtil::SplitSkipEmptyParts(proxy, ":");
-    if (ip_port.size() == 2) {
-        return {true, ip_port.at(0), ip_port.at(1)};
+    uint16_t port{0};
+    std::string hostname;
+    if (!SplitHostPort(proxy.toStdString(), port, hostname) || port != 0) {
```

Comment:
`SplitHostPort()` returns "true if port-portion is absent or within its allowed range". Here we want the port to be present and with its allowed range. So if we check that `SplitHostPort()` is true, that is not sufficient. For our usecase `SplitHostPort()` must be true and the port must be present. How to check whether the port is present? That would be `port != 0`. So that condition should be:

```suggestion
    if (SplitHostPort(proxy.toStdString(), port, hostname) && port != 0) {
        // valid and port is present and within the valid range
```
If there is a bug in `SplitHostPort()` that should be fixed separately.

Reply:
After playing with the function and the tests (`netbase_tests` - `TestSplitHost` - & `net_tests` - `addr.IsValid()`), what you propose looks more correct/ understandable to me as the use cases we were looking at were "::1:1080" (which passed to `SplitHostPort()` returns valid/ true and port 0) vs "[::1]:1080" (returning true and port 1080).

> If there is a bug in SplitHostPort() that should be fixed separately.

I think it's clear to me now that if you have multiple ":" it's because it's an ipv6 address and in that case you need to use the brackets to separate it from the port as [ipv6]:port otherwise the split will be ignored and the function will return the whole thing as the hostname (check other cases in `TestSplitHost`). Perhaps we can add these 2 cases in the tests so there's no confusion in the future.